lib/AST/ASTPrinter.cpp

```diff
@@ -4321,7 +4324,9 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
         Mod = Declaring;
     }
 
-    Identifier Name = Mod->getName();
+    // Should use the module real (binary) name here and everywhere else the
+    // module is printed in case module aliasing is used (see -module-alias)
+    Identifier Name = Mod->getRealName();
```

Comment:
Huh. These ASTPrinter cases are interesting, because I think some clients (like .swiftinterface printing) will want the real name and some (diagnostics, possibly some SourceKit clients) will want the alias name.

It might make sense to base this decision on the `PrintOptions`—either by putting a bool flag there, or by actually putting a map of aliases there.

(It’s probably okay to merge this as-is for now and come back to it later.)